[pruner] modify base pruner to prune bias by default

kazhou opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #63202
* #63178
* #63158

By default, the prune will also prune biases, such that the whole output channel is removed. The user can manually set `also_prune_bias` to False when calling `prepare` if they don't want the bias to be pruned.

Differential Revision: [D30294494](https://our.internmc.facebook.com/intern/diff/D30294494/)

kazhou committed:
[pruner] modify base pruner to prune bias by default

By default, the prune will also prune biases, such that the whole output channel is removed. The user can manually set `also_prune_bias` to False when calling `prepare` if they don't want the bias to be pruned.

Differential Revision: [D30294494](https://our.internmc.facebook.com/intern/diff/D30294494/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] modify base pruner to prune bias by default"


By default, the prune will also prune biases, such that the whole output channel is removed. The user can manually set `also_prune_bias` to False when calling `prepare` if they don't want the bias to be pruned.

Differential Revision: [D30294494](https://our.internmc.facebook.com/intern/diff/D30294494/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] modify base pruner to prune bias by default"


By default, the prune will also prune biases, such that the whole output channel is removed. The user can manually set `also_prune_bias` to False when calling `prepare` if they don't want the bias to be pruned.

Differential Revision: [D30294494](https://our.internmc.facebook.com/intern/diff/D30294494/)

[ghstack-poisoned]

raghuramank100 commented on torch/ao/sparsity/experimental/pruner/base_pruner.py:
```diff
@@ -124,16 +111,26 @@ def _prepare(self, use_path=False, *args, **kwargs):
 
-    def prepare(self, model, config):
-        r"""Prepares a model, by adding the parametrizations.
+    def prepare(self, model, config, also_prune_bias=True):
```
Does this need to be an argument for prepare? It will be cleaner to have this be part of the sparsifier definition, i,e lets specify this option during init.

raghuramank100 reviewed (changes requested):
Please see comments, looks almost ready to land!

kazhou committed:
Update on "[pruner] modify base pruner to prune bias by default"


By default, the prune will also prune biases, such that the whole output channel is removed. The user can manually set `also_prune_bias` to False when calling `prepare` if they don't want the bias to be pruned.

Differential Revision: [D30294494](https://our.internmc.facebook.com/intern/diff/D30294494/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] modify base pruner to prune bias by default"


By default, the prune will also prune biases, such that the whole output channel is removed. The user can manually set `also_prune_bias` to False when calling `prepare` if they don't want the bias to be pruned.

Differential Revision: [D30294494](https://our.internmc.facebook.com/intern/diff/D30294494/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] modify base pruner to prune bias by default"


By default, the prune will also prune biases, such that the whole output channel is removed. The user can manually set `also_prune_bias` to False when calling `prepare` if they don't want the bias to be pruned.

Differential Revision: [D30294494](https://our.internmc.facebook.com/intern/diff/D30294494/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] modify base pruner to prune bias by default"


By default, the prune will also prune biases, such that the whole output channel is removed. The user can manually set `also_prune_bias` to False when calling `prepare` if they don't want the bias to be pruned.

Differential Revision: [D30294494](https://our.internmc.facebook.com/intern/diff/D30294494/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] modify base pruner to prune bias by default"


By default, the prune will also prune biases, such that the whole output channel is removed. The user can manually set `also_prune_bias` to False when calling `prepare` if they don't want the bias to be pruned.

Differential Revision: [D30294494](https://our.internmc.facebook.com/intern/diff/D30294494/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] modify base pruner to prune bias by default"


By default, the prune will also prune biases, such that the whole output channel is removed. The user can manually set `also_prune_bias` to False when calling `prepare` if they don't want the bias to be pruned.

Differential Revision: [D30294494](https://our.internmc.facebook.com/intern/diff/D30294494/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] modify base pruner to prune bias by default"


By default, the prune will also prune biases, such that the whole output channel is removed. The user can manually set `also_prune_bias` to False when calling `prepare` if they don't want the bias to be pruned.

Differential Revision: [D30294494](https://our.internmc.facebook.com/intern/diff/D30294494/)

[ghstack-poisoned]
